### ISSUE TITLE
[oracle] Missing expression support in FOR loop ranges (length(x) and collection methods)

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8072]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example below leads to a 'Found unparsable section' error.

### Expected Behaviour

The example should be parsed correctly.

### Observed Behaviour

Linting the example below gives the following error:
`Line 1, Position 1: Found unparsable section: 'begin/n    for i in 1..length(10) loop/n  ...`
and
`Line 1, Position 1: Found unparsable section: 'begin/n    for i in 1..smth.count() loop/n  ...`
(This is the same for other collection methods as well (see https://docs.oracle.com/en/database/oracle/oracle-database/26/lnpls/collection-methods.html)

### How to reproduce

```sql
begin
    for i in 1..length(10) loop
           dbms_output.put_line(i);
    end loop;
end;

``` 

```sql
begin
    for i in 1..smth.count() loop
           dbms_output.put_line(i);
    end loop;
end;

``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
